File: src/main/java/com/nanaios/polygonal_tech/container/impl/IOMode.java
package com.nanaios.polygonal_tech.container.impl;

import com.nanaios.polygonal_tech.container.interfaces.IIOMode;

public class IOMode {
    public static final IIOMode INPUT = new Impl("input");
    public static final IIOMode OUTPUT = new Impl("output");
    public static final IIOMode INPUT_OUTPUT = new Impl("input_output");
    public static final IIOMode NONE = new Impl("none");

    public record Impl(String name) implements IIOMode { }
}

File: src/main/kotlin/com/nanaios/polygonal_tech/core/capability/builder/IOMode.kt
package com.nanaios.polygonal_tech.core.capability.builder

enum class IOMode(
    val canInput:Boolean,
    val canOutput:Boolean,
) {
    NONE(false, false),
    INPUT(true, false),
    OUTPUT(false, true),
    INPUT_OUTPUT(true, true);
}